packages/jest-core/src/testSchedulerHelper.ts

```diff
@@ -27,25 +27,25 @@ export function shouldRunInBand(
   }
 
   /*
-   * Run in band if we only have one test or one worker available, unless we
-   * are using the watch mode, in which case the TTY has to be responsive and
-   * we cannot schedule anything in the main thread. Same logic applies to
-   * watchAll.
+   * If we are using watch/watchAll mode, don't schedule anything in the main
+   * thread to keep the TTY responsive and to keep the watcher running even if
+   * the test crashes.
```

Comment:
Hm.. How to be more precise here? It isn’t that workers do prevent crashing tests, that is more about preventing leaks / open handles or debugging them (as you pointed out in the issue).

There is no mechanism to deal with this problem if test run in band. Leaks are handled more gracefully in parallel run, because worker farm gets killed after tests run is finished:

https://github.com/facebook/jest/blob/470d72ee74e50291bf8593954f24eb75b5621351/packages/jest-runner/src/index.ts#L180-L194

Reply:
Thanks for the suggestions -- took another stab at explaining this.